serverjwtauth/README.md

```diff
@@ -0,0 +1,113 @@
+[RFC 7517](https://datatracker.ietf.org/doc/html/rfc7517)). Once the key's signature has been validated, the
+`permissions` claim in the token is used to determine what the caller can do. See
+[Temporal security documentation](https://docs.temporal.io/docs/server/security) for more details.
+
+This example creates a ECDSA key pair, serves the public key for use by the server, and creates keys as needed for
+using `tctl`, running workers, and starting workflows.
+
+Note: Since we are not authorizing the UI in this sample, the server UI will not work with this restricted access.
+[Configuration of SSO for the Temporal Web UI](https://github.com/temporalio/web#configuring-authentication-optional) is
+outside the scope of this example.
+
+## Running
+
+### Starting the server
+
+First, the public keys must be served for reading by the server. Run the following in a separate terminal to generate a
+private key and serve its public key:
+
+    go run ./serverjwtauth/key gen-and-serve
+
+This will output a line like:
+
+    Started JWKS server. Endpoint: http://[::]:61884/jwks.json. Ctrl+C to exit.
+
+This creates a file at `key.priv.pem` with the private key. The port listed on each run will likely be different, so
+replace `61884` with the output port in all commands henceforth.
+
+With this running we can start the server. See the primary README about starting the server.
+[docker-compose](https://github.com/temporalio/docker-compose) is suggested for this sample.
+
+Before starting the server you must configure authorization. If using the docker image, the following environment
+variables should be set:
+
+* `TEMPORAL_JWT_KEY_SOURCE1=http://host.docker.internal:61884/jwks.json`
+* `TEMPORAL_AUTH_AUTHORIZER=default`
+* `TEMPORAL_AUTH_CLAIM_MAPPER=default`
+
+If using the docker-compose local install, these can be set in `docker-compose.yml`. If using a standalone server
+locally, the config is:
+
+```yaml
+global:
+  authorization:
+    jwtKeyProvider:
+      keySourceURIs:
+      - http://127.0.0.1:61884/jwks.json
+      refreshInterval: 1m
+    authorizer: default
+    claimMapper: default
+```
+
+With these set, start the server in the background via whatever install method chosen.
+
+### Using `tctl` and registering the `default` namespace
+
+When starting the server using the docker container, the following excerpt will appear in the logs:
+
+    "msg":"uncategorized error","operation":"RegisterNamespace","wf-namespace":"default","error":"Request unauthorized."
+
+This is because by default the docker container attempts to register the `default` namespace, but cannot because we have
+restricted access to the server to only authorized uses.
+
+To create this manually, we will use `tctl`. According to
+[the tctl docs](https://docs.temporal.io/docs/devtools/tctl/#securing-tctl) we can use the `tctl-authorization-plugin`
+binary. If using docker, this binary is included with tctl otherwise, built it and put it on the `PATH`.
```

Comment:
```suggestion
binary. If using docker, this binary is included with tctl. Otherwise, build it and put it on the `PATH`.
```

Reply:
Note that the homebrew install of `tctl` also installs this plugin. The only time someone would need to build and install the plugin is if they are running a tctl they have compiled themselves which seems a bit of an edge case for working from a sample.

Reply:
I don't use mac and a lot of people don't use mac and sadly we don't provide executables for other platforms.